server/realTime/scripts/Network.js

```diff
@@ -202,29 +220,68 @@ Protocol.prototype.getRegionSearchFromSat = function getRegionSearchFromSat (sat
 	}
 };
 
-Protocol.prototype.isInsideProtectedRegion = function isInsideProtectedRegion (owner, satObjects, room) {
+Protocol.prototype.getUsersProtectedRegions = function getUsersProtectedRegions (user, room) {
 	if (!this.protectedRegions[room]) return false;
 
+	var p = [];
+	var protectedRegionsArr = this.protectedRegions[room].all();
+
+	if(protectedRegionsArr.length == 0) return false;
+
+	for (var i = protectedRegionsArr.length - 1; i > 0; i--) {
+		//if (typeof protectedRegionsArr[i].permissions !== "undefined")
+			if( protectedRegionsArr[i].owner === user)
+				p.push({ 
+					regionId: protectedRegionsArr[i].id,
+					owner: protectedRegionsArr[i].owner, 
+					permissions: protectedRegionsArr[i].permissions || [],
+					minX: protectedRegionsArr[i].minX,
+					minY: protectedRegionsArr[i].minY,
+					maxX: protectedRegionsArr[i].maxX,
+					maxY: protectedRegionsArr[i].maxY,
+					minRepAllowed: protectedRegionsArr[i].minRepAllowed
+				});
+	}
+	return p;
+};
+
+Protocol.prototype.isInsideProtectedRegion = function isInsideProtectedRegion (reputation, user, satObjects, room) {
+	if (!this.protectedRegions[room]) return {isNotAllowed: false};
```

Comment:
You generally should not write variables or functions with negatives. It should be isAllowed: true, isNotAllowed: false is very hard to understand. ;)